src/blueapi/cli/cli.py

```diff
@@ -37,11 +36,21 @@
 @click.option(
     "-c", "--config", type=Path, help="Path to configuration YAML file", multiple=True
 )
+@click.option("--env-prefix", default="APP_", help="Prefix for environment variables")
 @click.pass_context
-def main(ctx: click.Context, config: Path | None | tuple[Path, ...]) -> None:
-    # if no command is supplied, run with the options passed
+def main(
+    ctx: click.Context, config: Path | None | tuple[Path, ...], env_prefix: str
+) -> None:
+    """
+    Main entry point for the CLI,
+    loading configuration in the correct order of precedence:
+    1. Defaults -> 2. File -> 3. Environment variables -> 4. CLI arguments.
```

Comment:
Nit: This will be displayed to the user when they type `blueapi --help` so maybe make it a bit more user-facing

```suggestion
    Blueapi CLI, see below for options.
    The configuration is loaded in the following order:
    1. Defaults
    2. File (via -c option)
    3. Environment variables (prefixed with {prefix})
    4. CLI arguments <<explain the format you expect>>
```

Point 4. is especially important because I can't quite tell from reading this code how I should pass config overrides in via the CLI.

Reply:
```python
    """
    BlueAPI CLI, see below for options.
    The configuration is loaded in the following order,
    each step overriding the previous one:
      1. Hardcoded defaults
      2. File (via -c option)
      3. Environment variables (prefixed with BLUEAPI_)
      4. CLI arguments (in a dot format like `--app.logging.level=DEBUG`)
    """
    
    ```
    
    
this as good enough - verified by @DiamondJoseph